ble/Gap.h

```diff
@@ -140,10 +140,15 @@ class Gap {
         return (durationInMillis * 1000) / UNIT_1_25_MS;
     }
 
+    typedef FunctionPointerWithContext<TimeoutSource_t> TimeoutEventCallback_t;
+    typedef CallChainOfFunctionPointersWithContext<TimeoutSource_t> TimeoutEventCallbackChain_t;
+
+    typedef FunctionPointerWithContext<const ConnectionCallbackParams_t *> ConnectionEventCallback_t;
+    typedef CallChainOfFunctionPointersWithContext<const ConnectionCallbackParams_t *> ConnectionEventCallbackChain_t;
```

Comment:
From my understanding, you want an inner type in callbackChain_t which define the type of the callback stored by the callback chain, is it what you want ? 

If so I have started to make the change but it make the declaration more complicated and exposes details of the implementation (the call chain type).

For instance, in Gap:

```c++
void onConnection(const ConnectionEventCallback_t& callback);
```

became 

```C++
    void onConnection(const ConnectionEventCallbackChain_t::FunctionPointerWithContext_t& callback);
```
And other part of the program (like BLE.h) which were using `Gap::ConnectionEventCallback_t` should now use `Gap::ConnectionEventCallbackChain_t::FunctionPointerWithContext_t` .

Is it what you want ?